src/timeseriesflattener/feature_spec_objects.py

```diff
@@ -66,6 +59,33 @@ def resolve_values_df(data: dict[str, Any]):
                 feature_name=data["feature_name"],
             )
 
+
+def resolve_values_df(data: dict[str, Any]):
+    """Resolve the values_df attribute to a dataframe."""
+    if (
+        "values_loader" not in data
+        and "values_df_dict" not in data
+        and "values_df" not in data
+    ):
+        raise ValueError(
+            "Either values_loader or a dictionary containing dataframes or a single dataframe must be specified.",
+        )
+
+    if (
+        in_dict_and_not_none(d=data, key="values_loader")
+        and in_dict_and_not_none(d=data, key="values_df_dict")
+        and in_dict_and_not_none(
+            key="values_df",
+            d=data,
+        )
+    ):
```

Comment:
Oh, this will introduce a bug. It only raises the error if all of them are in dict and not none, but we want it to raise an error if 2 or more are in the dict and not none.

```suggestion
    if sum(in_dict_and_not_none(data, key) for key in ["values_loader", "values_df_dict", "values_df"]) > 1:
```

Reply:
Would be great to add a regression test here - basically just create a spec that specifies two of them and show that it raises the ValueError :-)